compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1609,6 +1609,7 @@ public void visit(BLangTupleVariable varNode, AnalyzerData data) {
         typeChecker.checkExpr(varNode.expr, currentEnv, varNode.getBType(), data.prevEnvs,
```

Comment:
Can we check this for other binding patterns also? E.g., still no error for

```ballerina
public function main() {
    record {|
        int a;
    |} r = {a: 1};
    record {|
        @UndefinedAnnotation int a;
    |} {a} = r;
}
```

Assuming they are similar minor changes.

Reply:
Updated